#
# Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gnome-media/+bug/299642
# Upstream: http://bugzilla.gnome.org/show_bug.cgi?id=578174
# Description: Symptom is on startup all record toggles for capture tracks are set to disabled
#
diff --git a/gst-mixer/src/button.c b/gst-mixer/src/button.c
index 4cc65dc..eb84111 100644
--- a/gst-mixer/src/button.c
+++ b/gst-mixer/src/button.c
@@ -97,6 +97,8 @@ void
 gnome_volume_control_button_set_active (GnomeVolumeControlButton *button,
 					gboolean active)
 {
+  active = active != FALSE;
+
   if (button->active != active)
     gtk_button_clicked (GTK_BUTTON (button));
 }
